tt_metal/impl/dispatch/dispatch_core_manager.hpp

```diff
@@ -108,14 +108,39 @@ class dispatch_core_manager {
         return assignment.command_dispatcher.value();
     }
 
-    const std::optional<tt_cxy_pair> &remote_processor_core(chip_id_t device_id, uint16_t channel, uint8_t cq_id) {
-        TT_THROW("Do not currently support programming remote processor dispatch core. See https://github.com/tenstorrent-metal/tt-metal/issues/3953");
-        return this->dispatch_core_assignments[device_id][channel][cq_id].remote_processor;
+    /// @brief Gets the location of the kernel that receives fast dispatch packets from an ethernet router core and transfers them to the `command_dispatcher_core`
+    ///         This core is only programmed when targeting fast dispatch on the remote chip and its location is on the same chip that runs the fast dispatch commands
+    /// @param device_id ID of the remote device that will run fast dispatch commands
+    /// @param channel assigned to the command queue where commands targeting the device are enqueued
+    /// @param cq_id ID of the command queue within the channel
+    /// @return tt_cxy_pair logical location (chip + core coordinate) of the remote processor core
```

Comment:
thinking ahead here, when we call `get_associated_mmio_device` we should either query from the `CHAIN` of chips that we define. Or have associated mmio devices be the `CHAIN`. Can you add a TODO to cluster's `this->devices_grouped_by_assoc_mmio_device_`?

Reply:
sorry what do you mean by "associated mmio devices be the `CHAIN`"?

Reply:
Currently, we group the devices to assoc mmio device using UMD's `get_closest_link` api. We can make another api, like `group_to_fast_dispatch_chains(int num_mmio_chips)`. Then we should group the associated mmio devices using this api? Or whenever we call `get_associated_mmio_device` we can query some other object other than the `devices_grouped_by_assoc_mmio_device_`, but it seems like we should just keep those as the same object.

Reply:
ohh yes I see what you mean. For a MMIO device + its collection of remote devices there can be multiple "FD chains" 

We should just uplift to not use the `get_closest_link` and rather define our own collection of remote devices for a given MMIO device

Reply:
right, otherwise it's messy. Like are we expecting to use different drivers between chips in one chain?

Reply:
I don't think so